src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -76,11 +76,13 @@ public override ITestEngineRunner MakeTestRunner(TestPackage package)
                     projectCount++;
             }
 
+            ProcessModel processModel = GetTargetProcessModel(package);
+
             // If we have multiple projects or a project plus assemblies
             // then defer to the AggregatingTestRunner, which will make
             // the decision on a file by file basis so that each project
             // runs with its own settings.
-            if (projectCount > 1 || projectCount > 0 && assemblyCount > 0)
+            if (projectCount > 1 || (projectCount > 0 && assemblyCount > 0) || (projectCount > 0 && processModel == ProcessModel.Default))
```

Comment:
> OTOH, some tests that are slightly more integrated than what we have already, run as a part of the unit tests, would fit in well here.

Okay, my terminology may be off. It's unit testing but it's testing more than one layer at a time.